pkg/cli/cli.go

```diff
@@ -53,9 +57,34 @@ func CreateRole() error {
 	return nil
 }
 
-// TODO (fanz) : Create vSphere solution user (generate keypair), to be used with CCM
-func CreateSolutionUser(o ClientOption, c *govmomi.Client) error {
-	return nil
+// CreateSolutionUser creates a default solution user (k8s-vcp) for CCM with Administrator role and WSTrust permissions
+func CreateSolutionUser(ctx context.Context, o *ClientOption) error {
+	u := User{}
+	return u.Run(ctx, o, CreateUserFunc(func(c *ssoadmin.Client) error {
+		if cert, err := ReadContent(o.getCredential().getCertificate()); err == nil {
+			block, _ := pem.Decode([]byte(cert))
+			if block != nil {
+				u.solution.Certificate = base64.StdEncoding.EncodeToString(block.Bytes)
+			}
+			u.solution.Certificate = cert
+		}
+		if u.solution.Certificate == "" {
+			return fmt.Errorf("Need solution user certificate (--cert) to create solution user\n")
+		}
+		u.solution.Description = u.AdminPersonDetails.Description
+		if err := c.CreateSolutionUser(ctx, u.id, u.solution); err != nil {
+			return err
+		}
+		p := types.PrincipalId{Name: "k8s-vcp", Domain: c.Domain}
+
+		if _, err := c.SetRole(ctx, p, "Administrator"); err != nil {
```

Comment:
Role should probably be a flag, we can default to `Administrator`.  A separate feature for vcpctl could be to create a role with the minimum set of privileges required by VCP.

Reply:
Yes, added a flag for role (default is Administrator). 
`Create Role` could be in separate PR.